### PR TITLE
[NUI] Add line geometry APIs

### PR DESCRIPTION
**Pull request by saljammal**
### Description of Change ###
Retrieves the size and the position of a line. The below functions are added to the TextGeometry.cs file:
* GetLinePosition
* GetLineSize

### API Changes ###

Added:
  - public static List<Size2D> GetLineSize(TextLabel textLabel, int number) //Method
  - public static List<Size2D> GetLineSize(TextEditor textEditor, int number) //Method
  - public static List<Size2D> GetLineSize(TextField textField, int number) //Method
  - public static List<Position2D> GetLinePosition(TextEditor textEditor, int number) //Method
  - public static List<Position2D> GetLinePosition(TextField textField, int number) //Method
  - public static List<Position2D> GetLinePosition(TextLabel textLabel, int number) //Method

Depends on the following patches: 
- https://review.tizen.org/gerrit/c/platform/core/uifw/dali-toolkit/+/278414
- https://review.tizen.org/gerrit/c/platform/core/uifw/dali-csharp-binder/+/279260

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs [hunk 1]
```diff
@@ -212,5 +216,137 @@ public static List<Position2D> GetTextPosition(TextLabel textLabel, int start, i
...
+
+        /// <summary>
+        /// Get the rendered size of the line based on a requested value, zero-based. <br />
+        /// if a line contains characters with different directions, multiple sizes will be returned for each block of contiguous characters with the same direction. <br />
+        /// </summary>
+        /// <param name="textEditor">The TextEditor control containing the text.</param>
+        /// <param name="number">The number of the line to get the size for, zero-based.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static List<Size2D> GetLineSize(TextEditor textEditor, int number)
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs:228]: We prefer interface, 
```suggestion
        public static IList<Size2D> GetLineSize(TextEditor textEditor, int number)
```

Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs:228]: And How about declared as extensions method?
```suggestion
        public static List<Size2D> GetLineSize(this TextEditor textEditor, int number)
```

``` c#
  var size = textEditor.GetLineSize(1);
```

File: src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs [hunk 2]
```diff
@@ -28,12 +28,16 @@ namespace Tizen.NUI.BaseComponents
     [EditorBrowsable(EditorBrowsableState.Never)]
     public static class TextGeometry
     {
+        private static void ValidateIndex(int index)
+        {
+            if (index < 0)
+                throw new global::System.ArgumentOutOfRangeException(nameof(index), "Value is less than zero");
+        }
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs:35]: If index was exceed if the number of lines, what is happen?

File: src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs [hunk 3]
```diff
@@ -212,5 +216,137 @@ public static List<Position2D> GetTextPosition(TextLabel textLabel, int start, i
             CheckSWIGPendingException();
             return list;
         }
+
+        /// <summary>
+        /// Get the rendered size of the line based on a requested value, zero-based. <br />
+        /// if a line contains characters with different directions, multiple sizes will be returned for each block of contiguous characters with the same direction. <br />
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs:222]: I wonder about return value, 
Could you explain more details with some example?
I don't know means of `different directions`
and each item of returning value is corresponding each characters?
and if it has same appearance, is only returning one value?

File: src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs [hunk 4]
```diff
@@ -212,5 +216,137 @@ public static List<Position2D> GetTextPosition(TextLabel textLabel, int start, i
...
+        /// </summary>
+        /// <param name="textField">The textField control containing the text.</param>
+        /// <param name="number">The number of the line to get the size for, zero-based.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+         [EditorBrowsable(EditorBrowsableState.Never)]
+        public static List<Size2D> GetLineSize(TextField textField, int number)
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs:250]: How about lineNumber as parameter name, It is a more explained name.
```suggestion
        public static List<Size2D> GetLineSize(TextField textField, int lineNumber)
```

**Comment by saljammal** [comment]
Greetings, thank you @myroot for reviewing the pull request. Your notes will be considered. 
This request will be closed as there are some major changes in the requirements.

**Pull request closed without merging**